### ISSUE TITLE
How long does it need to train a cifar-10 model with FID 3.5?

### ISSUE THREAD

**Issue by glynpu** [atong01/conditional-flow-matching#105]
Dear all,
Thanks for the great package!

I am training with suggested command:
```
python3 train_cifar10.py --model "otcfm" --lr 2e-4 --ema_decay 0.9999 --batch_size 128 --total_steps 400001 --save_step 20000
```
But at a rough estimation, it needs 3 days(60 hours) to finish the whole training process with log:
```
lr, total_steps, ema decay, save_step: 0.0002 400001 0.9999 20000
Files already downloaded and verified
Model params: 34.09 M
  0%|▍                  | 1255/400001 [11:11<60:19:15,  1.84it/s]
```

I am using a 32GB V100.  I have two questions about training speed:
1. Is my current speed normal? (i.e. 60hours on a single V100 to train default 400001 batches)
2. Will results become worse if I double the default batch-size. Since I notice only half of the gpu memory is used now. (15424 / 32768MB)

**Comment by ImahnShekhzadeh** [comment]
Hi,
When I trained `train_cifar10.py` on a 4090 (also with the `otcfm` model) the training time took about 19 hours, and then sampling, for which you need to run
```
python3 compute_fid.py --model "otcfm" --step 400000 --integration_method dopri5
```
afterwards, about 30 minutes. The final FID I obtained was about 3.8, but since the weight initialization is random, small deviations are expected.

Since the 4090 is a better GPU than the V100, your obtained speed can indeed be normal. You can double the batch size (I had only trained with a batch size of `128`) to utilize full GPU memory, but then you should also adjust the learning rate accordingly (and perhaps also `ema_decay`, not quite sure about that).

**Comment by kilianFatras** [comment]
I agree, if you change the batch size and reduce the number of iterations, I suggest playing with the ema_decay variable as well. Unfortunately, I do not know in which magnitude you should reduce it…
